src/SigninResponse.ts

```diff
@@ -29,31 +29,31 @@ export class SigninResponse {
 
     // set by ResponseValidator
     /** custom "state", which can be used by a caller to have "data" round tripped */
-    public state: unknown | undefined;
+    public state: unknown;
 
     // set by ResponseValidator
-    public profile: UserProfile;
-
-    public constructor(url?: string, delimiter = "#") {
-        const values = UrlUtils.parseUrlFragment(url, delimiter);
-
-        this.error = values.error;
-        this.error_description = values.error_description;
-        this.error_uri = values.error_uri;
-
-        this.code = values.code;
-        this.state_id = values.state;
-
-        this.id_token = values.id_token;
-        this.session_state = values.session_state;
-        this.access_token = values.access_token;
-        this.refresh_token = values.refresh_token;
-        this.token_type = values.token_type;
-        this.scope = values.scope;
-        this.expires_in = parseInt(values.expires_in);
-
-        this.state = undefined;
-        this.profile = {};
+    public profile: UserProfile = {};
+
+    public constructor(params: URLSearchParams) {
+        // URLSearchParams returns `null` for missing values, reconstruct it as a map
+        const values = new Map(params);
+        this.error = values.get("error");
+        this.error_description = values.get("error_description");
+        this.error_uri = values.get("error_uri");
+
+        /* eslint-disable @typescript-eslint/no-non-null-assertion */
+        this.code = values.get("code")!;
+        this.access_token = values.get("access_token")!;
+        this.token_type = values.get("token_type")!;
+        /* eslint-enable */
+
+        this.state_id = values.get("state");
+        this.id_token = values.get("id_token");
+        this.session_state = values.get("session_state");
+        this.refresh_token = values.get("refresh_token");
+        this.scope = values.get("scope");
+        const expiresIn = values.get("expires_in");
+        this.expires_in = expiresIn == null ? undefined : parseInt(expiresIn);
```

Comment:
expiresIn can not be null, but undefined
->
this.expires_in = parseInt(expiresIn || "");

parseInt("") gives undefined...

Reply:
I didn't realize that `parseInt` had that behavior, I'll update this

Reply:
`parseInt("")` actually yields `NaN` which I don't think is what we want. I think the code I have is the soundest option still (expiresIn is nullish -> leave undefined ; else parse as integer).

Reply:
you are right the console tricked me :-( argh

still expiresIn can not be null, why not simple:
`this.expires_in = expiresIn ? parseInt(expiresIn) : undefined;`

Reply:
Yeah that's fine actually, I forgot that we don't have to worry about properly handling the `0` case.